presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionSymbolInliner.java

```diff
@@ -40,9 +45,15 @@ public Expression rewrite(Expression expression)
     private class Visitor
             extends ExpressionRewriter<Void>
     {
+        private final Set<String> excludedNames = new HashSet<>();
+
         @Override
         public Expression rewriteSymbolReference(SymbolReference node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
         {
+            if (excludedNames.contains(node.getName())) {
```

Comment:
When this will be the case? Is it legal to leave symbols in lambdas that should be rewritten outside of lambdas?
E.g:
```
Filter (with lambda using symbol X)
 |
Projection (produces X)
```
after `PredicatePushDown`
```
Projection (produces X)
 |
Filter (with lambda - X should be replaced with X')
```
Can it be the case?